### PR TITLE
Add more dartdoc to material.dart

### PR DESCRIPTION
**Pull request by abarth**
The dartdoc will continue until morale improves.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @HansMuller

File: packages/flutter/lib/src/material/checkbox.dart [hunk 1]
```diff
@@ -30,10 +30,10 @@ import 'toggleable.dart';
 ///  * <https://www.google.com/design/spec/components/selection-controls.html#selection-controls-checkbox>
 ///  * <https://www.google.com/design/spec/components/lists-controls.html#lists-controls-types-of-list-controls>
 class Checkbox extends StatelessWidget {
-  /// Constructs a checkbox
+  /// Creates a material design checkbox
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/checkbox.dart]: I believe that we punctuate anything that looks remotely like a sentence.

File: packages/flutter/lib/src/material/checkbox.dart [hunk 2]
```diff
@@ -30,10 +30,10 @@ import 'toggleable.dart';
...
   ///
-  /// * `value` determines whether the checkbox is checked.
-  /// * `onChanged` is called whenever the state of the checkbox should change.
+  /// * [value] determines whether the checkbox is checked.
+  /// * [onChanged] is called whenever the state of the checkbox should change.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/checkbox.dart]: Whenever the -value- of the checkbox should change?

Developers, particularly the ones who've used non-React style toolkits, may be surprised to discover that it's up to them to maintain the checkbox's value.  Maybe we should say as much after "Creates a .. checkbox".

File: packages/flutter/lib/src/material/chip.dart [hunk 3]
```diff
@@ -35,6 +35,9 @@ const TextStyle _kLabelStyle = const TextStyle(
 ///  * [CircleAvatar]
 ///  * <https://www.google.com/design/spec/components/chips.html>
 class Chip extends StatelessWidget {
+  /// Creates a material design chip.
+  ///
+  ///  * [onDeleted] determines whether the chip has a delete button.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/chip.dart]: This also gets called when the delete button is pressed, so maybe we should state the obvious. Something like:

If specified the chip has a delete button. This callback runs when the delete button is pressed.

File: packages/flutter/lib/src/material/constants.dart [hunk 4]
```diff
@@ -2,39 +2,20 @@
...
-import 'package:flutter/widgets.dart';
-
-// TODO(eseidel) Toolbar needs to change size based on orientation:
-// http://www.google.com/design/spec/layout/structure.html#structure-app-bar
-// Mobile Landscape: 48dp
-// Mobile Portrait: 56dp
-// Tablet/Desktop: 64dp
+/// The height of the tool bar component of the [AppBar].
 const double kToolBarHeight = 56.0;
-const double kExtendedAppBarHeight = 128.0;
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/constants.dart]: NICE. This should have been deleted some time ago.

File: packages/flutter/lib/src/material/date_picker.dart [hunk 5]
```diff
@@ -29,6 +29,10 @@ enum _DatePickerMode { day, year }
 ///  * [showDatePicker]
 ///  * <https://www.google.com/design/spec/components/pickers.html#pickers-date-pickers>
 class DatePicker extends StatefulWidget {
+  /// Creates a date picker.
+  ///
+  /// Rather than creating a date picker directly, cosider using
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/date_picker.dart]: cosider => consider

File: packages/flutter/lib/src/material/divider.dart [hunk 6]
```diff
@@ -6,13 +6,42 @@ import 'package:flutter/widgets.dart';
...
+/// [ListItem.divideItems], which is optimized for this case.
+///
+/// See also:
+///  * [ListItem.divideItems]
+///  * <https://www.google.com/design/spec/components/dividers.html>
 class Divider extends StatelessWidget {
-  Divider({ Key key, this.height: 16.0, this.indent: 0.0, this.color }) : super(key: key) {
+  /// Creates a material design divider.
+  ///
+  /// The hieght must be at least 1.0 logical pixels.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/divider.dart]: height

File: packages/flutter/lib/src/material/divider.dart [hunk 7]
```diff
@@ -6,13 +6,42 @@ import 'package:flutter/widgets.dart';
...
+  Divider({
+    Key key,
+    this.height: 16.0,
+    this.indent: 0.0,
+    this.color
+  }) : super(key: key) {
     assert(height >= 1.0);
   }
 
+  /// The amount of height the divider should occupy.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/divider.dart]: Suggestion: The divider's vertical extent.

File: packages/flutter/lib/src/material/divider.dart [hunk 8]
```diff
@@ -6,13 +6,42 @@ import 'package:flutter/widgets.dart';
...
+  ///
+  /// The divider itself is always drawn as one logical pixel thick horizontal
+  /// line that is centered within the height specified by this value.
   final double height;
+
+  /// The amount of empty space to the left and right of the divider.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/divider.dart]: I believe that indent only affects the divider's left side.

**Comment by HansMuller** [comment]
LGTM

File: packages/flutter/lib/src/material/checkbox.dart (hunk 2 above)
Comment (abarth) [inline packages/flutter/lib/src/material/checkbox.dart]: Yeah, we say that in the class description, but I'll replicate the text here.

File: packages/flutter/lib/src/material/divider.dart (hunk 7 above)
Comment (abarth) [inline packages/flutter/lib/src/material/divider.dart]: Done.

File: packages/flutter/lib/src/material/divider.dart (hunk 8 above)
Comment (abarth) [inline packages/flutter/lib/src/material/divider.dart]: Ah, so it does.

**Pull request merged**